library/network_connections.py

```diff
@@ -531,6 +531,16 @@ def ifcfg_create(
                 line = r["network"] + "/" + str(r["prefix"])
                 if r["gateway"]:
                     line += " via " + r["gateway"]
+                if connection["interface_name"]:
+                    line += " dev " + connection["interface_name"]
+                else:
+                    warn_fcn(
+                        "the route '{0}' from the connection '{1}' will be configured "
```

Comment:
The message is missing the reason for this and recommendations how to get rid of the warning:

"The connection ... does not specify an interface name. Therefore, the route ... will be configured without the output device and the kernel will choose it automatically which might result in an unwanted device being used. To avoid this, specify `interface_name` in the connection appropriately."

Reply:
Thanks for the suggestion.